### ISSUE TITLE
Request for Cosine Metric Learning

### ISSUE THREAD

**Issue by pvtien96** [KevinMusgrave/pytorch-metric-learning#63]
Hi, I'm a newbie to deep learning.
Currently I'm working on a tracking-by-detection project related to [deep sort](https://github.com/nwojke/deep_sort).
In order to customize deepsort for my own classes, I need to train [Cosine Metric Learning](https://github.com/nwojke/cosine_metric_learning).
I've searched around for a pytorch implementation of cosine metric learning on google and it leads me to your repo.
So I was wondering if this is the one that meets my requirement? 
If yes, how can I use cosine metric learning from your repo?
Many thanks!

**Comment by KevinMusgrave** [comment]
I took a quick look at the paper, and I think [NormalizedSoftmaxLoss](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#normalizedsoftmaxloss) is equivalent. The only difference is that the paper specifies a scaling parameter ```k```, whereas NormalizedSoftmaxLoss takes in ```temperature``` which is simply ```1/k```.

For this loss to work properly, you have to use it with an optimizer. For example:
```python
loss_func = losses.NormalizedSoftmaxLoss(temperature=0.1, embedding_size=128, num_classes=100)
loss_optimizer = torch.optim.SGD(loss_func.parameters(), lr=0.01)

# then during training:
loss_optimizer.step()
```

**Comment by pvtien96** [comment]
Yes, thanks for your quick response, I will try this.
+1 star for your repository :)
